src/main/scala/com/fulcrumgenomics/util/RefFlatSource.scala

```diff
@@ -0,0 +1,154 @@
+  *
+  * Skips genes on unrecognized chromosomes if a sequence dictionary is provided.
+  *
+  * The format is described here: http://genome.ucsc.edu/goldenPath/gbdDescriptionsOld.html#RefFlat
+  *
+  * The header should be uppercase if provided.
+  * */
+class RefFlatSource private(lines: Iterator[String],
+                            dict: Option[SAMSequenceDictionary] = None,
+                            private[this] val source: Option[{ def close(): Unit }] = None
+                           ) extends Iterable[Gene] with Closeable{
+  private val genes = {
+
+    // Ensure there is a header for DelimitedDataParser
+    val _lines = {
+      import com.fulcrumgenomics.FgBioDef._
+      val iterator = lines.bufferBetter
+      if (iterator.head.toUpperCase().startsWith("GENE_NAME")) iterator
+      else Iterator(RefFlatSource.Header) ++ iterator
+    }
+
+    // Each line is a gene and transcript.  We want to group all transcripts from the same gene.
+    new DelimitedDataParser(lines=_lines, delimiter='\t').flatMap { row =>
+      val geneName       = row[String]("GENE_NAME")
+      val transcriptName = row[String]("TRANSCRIPT_NAME")
+      val contig         = row[String]("CHROMOSOME")
+      val strand         = row[String]("STRAND")
+      val exonCount      = row[Int]("EXON_COUNT")
+      val exonStarts     = row[String]("EXON_STARTS").split(",").filter(_.nonEmpty).map(_.toInt)
+      val exonEnds       = row[String]("EXON_ENDS").split(",").filter(_.nonEmpty).map(_.toInt)
+
+      if (dict.exists(_.getSequence(contig) == null)) { // skip unrecognized sequences
+        None
+      }
+      else {
+        require(exonStarts.length == exonCount, s"Number of EXON_STARTS does not equal EXON_COUNT for $geneName/$transcriptName")
+        require(exonEnds.length == exonCount, s"Number of EXON_ENDS does not equal EXON_COUNT for $geneName/$transcriptName")
+
+        // Convert from 0-based half-open to 1-based inclusive intervals.
+        val transcript = Transcript(
+          name     = transcriptName,
+          txStart  = row[String]("TX_START").toInt + 1,
+          txEnd    = row[String]("TX_END").toInt,
+          cdsStart = row[String]("CDS_START").toInt + 1,
+          cdsEnd   = row[String]("CDS_END").toInt,
+          exons    = exonStarts.zip(exonEnds).map { case (s, e) => Exon(start=s.toInt + 1, end=e.toInt) }
+        )
+
+        val gene = Gene(
+          contig         = contig,
+          start          = -1,
+          end            = -1,
+          negativeStrand = strand == "-",
+          name           = geneName,
+          transcripts    = Seq(transcript)
+        )
+
+        Some(gene)
+      }
+    }
+      .toSeq
+      .groupBy(_.name)
+      .map { case (name, _genes) =>
+        val contig         = _genes.head.contig
+        val negativeStrand = _genes.head.negativeStrand
+        require(_genes.forall(_.contig == contig), s"Chromosomes differ for gene $name")
```

Comment:
This is going to hurt.  It's not unusual for gene databases at UCSC to have mappings to multiple chromosomes for a small fraction of genes.  If you throw an exception when you see this, you're going to have a bad time trying to clean up your refFlat data.

I'd suggest that instead, you either filter out those genes, or you only provide a single set of mappings, or you have a flag/def that tells you whether there are multiple contig mappings and return everything, so that clients can decide what to do about it.